toolchain/check/name_component.h

```diff
@@ -28,14 +28,16 @@ struct NameComponent {
 
   // The implicit parameter list.
   Parse::NodeId implicit_params_loc_id;
-  SemIR::InstBlockId implicit_params_id;
   SemIR::InstBlockId implicit_param_patterns_id;
 
   // The explicit parameter list.
   Parse::NodeId params_loc_id;
-  SemIR::InstBlockId params_id;
   SemIR::InstBlockId param_patterns_id;
 
+  // The calling convention parameters of the function. These will all be
```

Comment:
```suggestion
  // The calling convention parameters of the entity. These will all be
```

Won't this be set for all parameterized entities, not only functions?

Reply:
I believe functions are the only entities that have a calling convention (because they're the only entities that can be called with runtime arguments).

Reply:
Would it be reasonable to add a TODO similar to the one for the return slot fields here?

I think it'd also be useful to specify more explicitly what you mean by "calling convention parameters" here. Maybe overall something like:

> The calling convention parameters of this entity, if it is a function. These are the parameters that correspond to the arguments in a `Call` instruction that calls this function. These will all be instances of `AnyParam`.

Reply:
Hm, or maybe this should reference `EntityWithParamsBase` for the full explanation of this field. I think the TODO and the change from "the function" to "the entity if it is a function" would still be useful here, though.

Reply:
Done, and renamed to `call_params_id` as discussed at today's meeting.